app/src/main/java/org/thunderdog/challegram/data/TGMessageFile.java

```diff
@@ -695,4 +700,66 @@ protected boolean onLocaleChange () {
     }
     return changed;
   }
+
+  private TdApi.FormattedText getTranslationSafeText (TdApi.FormattedText text) {
+    if (translationStyleMode() == Settings.TRANSLATE_MODE_POPUP) return text;
+    return new TdApi.FormattedText(text.text.replaceAll("\uD83D\uDCC4", "\uD83D\uDCD1"), text.entities);
+  }
+
+  @Nullable
+  @Override
+  public TdApi.FormattedText getTextToTranslateImpl () {
+    if (filesList.size() == 1) {
+      CaptionedFile file = filesList.get(0);
+      return file.hasCaption() ? getTranslationSafeText(file.serverCaption): null;
+    }
+
+    TdApi.FormattedText resultText = new TdApi.FormattedText("", new TdApi.TextEntity[0]);
+    TdApi.FormattedText sep = new TdApi.FormattedText(translationStyleMode() == Settings.TRANSLATE_MODE_POPUP ? "\n\n": "\n\n\uD83D\uDCC4\n", new TdApi.TextEntity[0]);
+    int filesWithCaption = 0;
+
+    for (CaptionedFile file : filesList) {
+      if (file.hasCaption()) {
+        resultText = Td.concat(resultText, sep, getTranslationSafeText(file.serverCaption));
+        filesWithCaption++;
+      } else {
+        resultText = Td.concat(resultText, sep);
+      }
+    }
+
+    return filesWithCaption > 0? Td.trim(resultText): null;
+  }
```

Comment:
There is a slight side effect of this solution, which seems to be present only in the "inline" translation mode, that all 📄 emoji in captions to media files get replaced with 📑 emoji, which itself isn't too critical, but it's there even if it's just 1 file, not an album, in which case it seems `TranslateMessageText` method could have been used instead.